new metadata catalog column mime_type

fgrcon opened the pull request:
https://github.com/plone/Products.CMFPlone/issues/1995

gforcada commented on plone/app/upgrade/v51/configure.zcml:
```diff
@@ -133,11 +133,23 @@ Add image scaling options to image handling control panel.
+        <gs:upgradeDepends
+            title="Run to51beta4 upgrade profile."
+            description="
+New metadata column `mime_type`.
```
Just cosmetics, but is there any good reason to add the description in a new line?

fgrcon replied:
oops tabs instead of spaces! (fixed)

thet commented:
got an error while upgrading:

```
Starting the migration from version: 5106
Role / permission map imported.
Catalog imported.
Ran upgrade step: Run to51beta4 upgrade profile.
Fixed plone.app.caching.strongCaching.plone.resource.maxage to refer to plone.app.caching.strongCaching.maxage.
Ran upgrade step: Fix double shared maxage
Upgrade aborted. Error:
Traceback (most recent call last):
  File "/plone/buildout/src/Products.CMFPlone/Products/CMFPlone/MigrationTool.py", line 268, in upgrade
    step['step'].doStep(setup)
  File "/plone/buildout/eggs/Products.GenericSetup-1.8.6-py2.7.egg/Products/GenericSetup/upgrade.py", line 166, in doStep
    self.handler(tool)
  File "/plone/buildout/src/plone.app.upgrade/plone/app/upgrade/v51/betas.py", line 161, in reindex_mime_type
    new_value = obj.content_type()
  File "/plone/buildout/eggs/plone.app.contenttypes-1.2.22-py2.7.egg/plone/app/contenttypes/content.py", line 137, in content_type
    return self.image.contentType
AttributeError: 'NoneType' object has no attribute 'contentType'
End of upgrade path, main migration has finished.
The upgrade path did NOT reach current version.
Migration has failed
```

/cc @fgrcon

fgrcon commented:
@thet please help with easter egg hunt:
right now I have set up a fresh build of coredev 5.1  at https://github.com/plone/buildout.coredev/commit/f26df03cf5e56001b0074b8cd2ad74babca4784f and replaced filesorage and blobstorage with clean Plone 4.12 testdata:
```

Starting the migration from version: 4315
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Utilities registered.
Skins tool imported.
Archetype tool imported.
Properties tool imported.
Imported.
Step languagetool has an invalid import handler
Stylesheet registry imported.
Javascript registry imported.
Portlets imported
Cleaned up the toolset registry.
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Actions tool imported.
Control panel imported.
Archetype tool imported.
Browser layers imported
Step languagetool has an invalid import handler
Types tool imported.
'Folder' type info imported.
'Plone Site' type info imported.
'Collection' type info imported.
Stylesheet registry imported.
Javascript registry imported.
Cannot register portlet type portlets.Calendar because it is already registered.
Cannot register portlet type portlets.Events because it is already registered.
Portlets imported
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Skins tool imported.
Archetype tool imported.
Browser layers imported
Step languagetool has an invalid import handler
Theme barceloneta applied
Theme enabled
upgrading keyring
add keyring to zope root if not done already
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Control panel imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50alpha1 upgrade profile
Ran upgrade step: Use lowercase for email login
Ran upgrade step: Miscellaneous
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50alpha3 upgrade profile
Ran upgrade step: Upgrade editing control panel settings
Ran upgrade step: Upgrade maintenance control panel settings
Ran upgrade step: Upgrade navigation control panel settings
Ran upgrade step: Upgrade search control panel settings
Ran upgrade step: Upgrade site control panel settings
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Skins tool imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50beta1 upgrade profile
Ran upgrade step: Upgrade markup control panel settings
Ran upgrade step: Upgrade mail control panel settings
Ran upgrade step: Upgrade security control panel settings
Ran upgrade step: Remove Language Tool and migrate settings
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50beta2 upgrade profile
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Actions tool imported.
Control panel imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Removed record plone.resources/mockup-patterns-accessibility.init.
Removed record plone.resources/mockup-patterns-accessibility.export.
Removed record plone.resources/mockup-patterns-accessibility.deps.
Removed record plone.resources/mockup-patterns-accessibility.conf.
Removed record plone.resources/mockup-patterns-accessibility.url.
Removed record plone.resources/mockup-patterns-accessibility.css.
Removed record plone.resources/mockup-patterns-accessibility.js.
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50beta3 upgrade profile
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Actions tool imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Removed record plone.resources/mockup-patterns-tablesorter.init.
Removed record plone.resources/mockup-patterns-tablesorter.export.
Removed record plone.resources/mockup-patterns-tablesorter.deps.
Removed record plone.resources/mockup-patterns-tablesorter.conf.
Removed record plone.resources/mockup-patterns-tablesorter.url.
Removed record plone.resources/mockup-patterns-tablesorter.css.
Removed record plone.resources/mockup-patterns-tablesorter.js.
Removed record plone.resources/resource-collection-css.init.
Removed record plone.resources/resource-collection-css.export.
Removed record plone.resources/resource-collection-css.deps.
Removed record plone.resources/resource-collection-css.conf.
Removed record plone.resources/resource-collection-css.url.
Removed record plone.resources/resource-collection-css.css.
Removed record plone.resources/resource-collection-css.js.
Ran upgrade step: Run to50beta4 upgrade profile
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50beta4 upgrade profile
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Skins tool imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50rc1 upgrade profile
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to50rc2 upgrade profile
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Removed record plone.resources/resource-plone-app-discussion-stylesheets.init.
Removed record plone.resources/resource-plone-app-discussion-stylesheets.export.
Removed record plone.resources/resource-plone-app-discussion-stylesheets.deps.
Removed record plone.resources/resource-plone-app-discussion-stylesheets.conf.
Removed record plone.resources/resource-plone-app-discussion-stylesheets.url.
Removed record plone.resources/resource-plone-app-discussion-stylesheets.css.
Removed record plone.resources/resource-plone-app-discussion-stylesheets.js.
could not upgrade calendar_starting_year property
could not upgrade calendar_future_years_available property
Ran upgrade step: Run to50rc3 upgrade profile
Setting installed version of profile Products.PloneLanguageTool:PloneLanguageTool as unknown.
Setting installed version of profile plone.app.jquerytools:default as unknown.
Setting installed version of profile plone.app.jquery:default as unknown.
Setting installed version of profile Products.TinyMCE:TinyMCE as unknown.
Setting installed version of profile collective.z3cform.datetimewidget:default as unknown.
Setting installed version of profile archetypes.referencebrowserwidget:default as unknown.
Setting installed version of profile plone.app.search:default as unknown.
Setting installed version of profile archetypes.querywidget:default as unknown.
Setting installed version of profile Products.CMFPlone:plone-content as unknown.
Setting installed version of profile plonetheme.classic:default as unknown.
Setting installed version of profile plonetheme.sunburst:default as unknown.
Ran upgrade step: Unmark installed profiles that are no longer available.
Ran upgrade step: Mark products as installed for installed uninstallable profiles
Ran upgrade step: Cleanup uninstalled products
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to500 upgrade profile
Ran upgrade step: Unmark installed profiles that are no longer available.
Ran upgrade step: Mark products as installed for installed uninstallable profiles
Ran upgrade step: Cleanup uninstalled products
Removed no longer existing path mimetypes_icons from skin selection Plone Default.
Removed no longer existing path PasswordReset from skin selection Plone Default.
Removed no longer existing path plone_styles from skin selection Plone Default.
Removed no longer existing path plone_content from skin selection Plone Default.
Removed no longer existing path classic_styles from skin selection Plone Classic Theme.
Removed no longer existing path classic_ecmascript from skin selection Plone Classic Theme.
Removed no longer existing path classic_templates from skin selection Plone Classic Theme.
Removed no longer existing path mimetypes_icons from skin selection Plone Classic Theme.
Removed no longer existing path PasswordReset from skin selection Plone Classic Theme.
Removed no longer existing path plone_styles from skin selection Plone Classic Theme.
Removed no longer existing path plone_content from skin selection Plone Classic Theme.
Removed no longer existing path sunburst_images from skin selection Sunburst Theme.
Removed no longer existing path sunburst_templates from skin selection Sunburst Theme.
Removed no longer existing path sunburst_styles from skin selection Sunburst Theme.
Removed no longer existing path mimetypes_icons from skin selection Sunburst Theme.
Removed no longer existing path PasswordReset from skin selection Sunburst Theme.
Removed no longer existing path plone_styles from skin selection Sunburst Theme.
Removed no longer existing path plone_content from skin selection Sunburst Theme.
Ran upgrade step: Cleanup the skins tool.
Ran upgrade step: Remove fake kupu tool and related settings and resources.
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Actions tool imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Reindexed `getIcon` for 13 items
Ran upgrade step: Run to501 upgrade profile.
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to502 upgrade profile.
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to503 upgrade profile.
Ran upgrade step: Miscellaneous
Ran upgrade step: Miscellaneous
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to507 upgrade profile.
Ran upgrade step: Fix double shared maxage
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Control panel imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Interface Products.ResourceRegistries.interfaces.settings.IResourceRegistriesSettings specified for record %s has no field plone.lessvariables.
Ran upgrade step: Run to51alpha1 upgrade profile.
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Removed record plone.resources/mockup-registry.init.
Removed record plone.resources/mockup-registry.export.
Removed record plone.resources/mockup-registry.deps.
Removed record plone.resources/mockup-registry.conf.
Removed record plone.resources/mockup-registry.url.
Removed record plone.resources/mockup-registry.css.
Removed record plone.resources/mockup-registry.js.
Removed record plone.resources/mockup-parser.init.
Removed record plone.resources/mockup-parser.export.
Removed record plone.resources/mockup-parser.deps.
Removed record plone.resources/mockup-parser.conf.
Removed record plone.resources/mockup-parser.url.
Removed record plone.resources/mockup-parser.css.
Removed record plone.resources/mockup-parser.js.
Ran upgrade step: Run to51alpha2 upgrade profile.
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Toolset imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to51beta1 upgrade profile.
Added 'sort_on' property to site_properties.
Ran upgrade step: Add default search options
Ran upgrade step: Remove skins layer converted to browser resources
Ran upgrade step: Remove jquery.cookie from plone-logged-in stub_js_modules
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Toolset imported.
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to51beta1 upgrade profile.
Removed object/displayContentsTab from action at /Plone/portal_actions/object/folderContents
Ran upgrade step: Remove displayContentsTab from action expressions
Ran upgrade step: Miscellaneous
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Ran upgrade step: Run to51beta3 upgrade profile.
Field types updated on social media schema
Ran upgrade step: Update social media fields
Role / permission map imported.
Step collective.z3cform.datetimewidget has an invalid import handler
Archetype tool imported.
Step languagetool has an invalid import handler
Catalog imported.
Ran upgrade step: Run to51beta4 upgrade profile.
Ran upgrade step: Fix double shared maxage
Reindexed `mime_type` for 6 items
Ran upgrade step: Reindex mime_type
End of upgrade path, main migration has finished.
Starting upgrade of core addons.
Role / permission map imported.
Workflow tool imported.
Done upgrading core addons.
Your Plone instance is now up-to-date.
```
In case the problem remains, would you please provide some more information?
